aten/src/ATen/native/Sorting.cpp

```diff
@@ -171,7 +171,7 @@ std::tuple<Tensor&, Tensor&> kthvalue_out_cpu(
               tmp_values,
               k - 1,
               [](scalar_t x, scalar_t y) -> bool {
-                return ((x != x && y == y) || (x > y));
+                return ((std::isnan(x) && !std::isnan(y)) || (x > y));
```

Comment:
Does this work correctly and efficiently when scalar_t is an integral type?  I *think* this casts it to a double and still does a NaN check.

Reply:
Looks like you're right: https://godbolt.org/z/HhgWIu

Reply:
I believe you should be avale to use numeric traits to test if scalar_t is a floating point type.

Reply:
It looks like someone already wrote a function do this, I'm going to use it.